test/jdk/java/lang/StringBuilder/Insert.java

```diff
@@ -21,15 +21,31 @@
  * questions.
  */
 
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
 /**
  * @test
- * @bug 4914802
- * @summary Test Insert method for infinite loop
+ * @run testng Insert
+ * @bug 4914802 8257511
+ * @summary Test StringBuilder.insert sanity tests
  */
-
+@Test
 public class Insert {
-   public static void main (String argv[]) throws Exception {
-       StringBuilder sb = new StringBuilder();
-       sb.insert(0, false);
-   }
+
+    public void insertFalse() {
+        // Caused an infinite loop before 4914802
+        StringBuilder sb = new StringBuilder();
+        assertEquals("false", sb.insert(0, false).toString());
+    }
+
+    public void insertOffset() {
+        // 8254082 made the String variant cause an AIOOBE, fixed in 8257511
+        assertEquals("efabc", new StringBuilder("abc").insert(0, "def",                      1, 3).toString());
```

Comment:
The change looks okay.

Changing this test, and converting to TestNG is okay too. I guess I wouldn't try to align the start, end on L45 as it doesn't really help readability (doesn't matter of course).